Tools/AP_Bootloader/flash_from_sd.cpp

```diff
@@ -0,0 +1,385 @@
+#include "flash_from_sd.h"
+
+#if AP_BOOTLOADER_FLASH_FROM_SD_ENABLED
+
+#include "ch.h"
+#include "ff.h"
+
+#include "md5.h"
+
+#include <string.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include "stm32_util.h"
+
+#include <AP_HAL_ChibiOS/hwdef/common/flash.h>
+#include <AP_Math/AP_Math.h>
+#include "support.h"
+
+// swiped from support.cpp:
+static const uint8_t *flash_base = (const uint8_t *)(0x08000000 + (FLASH_BOOTLOADER_LOAD_KB + APP_START_OFFSET_KB)*1024U);
```

Comment:
This is not going to work with external flash.

Reply:
Noted.  Don't add the option on boards that have external flash for the time being.